ement-lib.el

```diff
@@ -772,14 +777,18 @@ THEN and ELSE are passed to `ement-api', which see."
     :content-type content-type :data data :data-type 'binary
     :then then :else else))
 
-(cl-defun ement-complete-session (&key (prompt "Session: "))
-  "Return an Ement session selected with completion."
+(cl-defun ement-complete-session (&key (pred #'identity) (prompt "Session: "))
+  "Return an Ement session selected with completion.
+PROMPT is passed to `completing-read'.  Only offer sessions
+matching PRED."
   (pcase (length ement-sessions)
-    (0 (user-error "No active sessions.  Call `ement-connect' to log in"))
-    (1 (cdar ement-sessions))
-    (_ (let* ((ids (mapcar #'car ement-sessions))
-              (selected-id (completing-read prompt ids nil t)))
-         (alist-get selected-id ement-sessions nil nil #'equal)))))
+    (0 (user-error "No sessions.  Call `ement-connect' to log in"))
+    (_ (let ((sessions (cl-remove-if-not pred ement-sessions :key #'cdr)))
+         (pcase (length sessions)
+           (1 (cdar sessions))
+           (_ (let* ((ids (mapcar #'car sessions))
+                     (selected-id (completing-read prompt ids nil t)))
+                (alist-get selected-id sessions nil nil #'equal))))))))
```

Comment:
I think a lot of the `ement-complete-session` uses could reasonably be expected to act on the session for the current room buffer.  How about we make that call to `completing-read` specify the id for `ement-session` as its `DEF`ault argument?

We'd then want to make the prompt `(format "%s [%s]: " prompt id)` in the case when there was a default.

Reply:
I was also wondering whether there could be cases in practice where the user thought they were in multiple sessions, but in fact only one of them was connected, and the "don't prompt when only one session is connected" behaviour caused them to do something in the wrong session.

It might be reasonable to have a user option to say "prompt whenever there are multiple sessions, even if only one of them is connected".